### ISSUE TITLE
App won´t authenticate in Harvest after update to Jira 8

### ISSUE THREAD

**Issue by shernandez-ctrlQS** [Strategery-Inc/jira-harvest-time-tracker#98]
Hi Strategry team,

We have just updated our Jira to 8.2.4 and the app to 1.0.13-SNAPSHOT, which is meant to be compatible with Jira 8.
The app won´t pass now the authentication step after approving the conneciton to our Harvest instance, I have reviewed the Harvest OAuth configuration (I even created a new one as well) and the App configuration and they look good.
I could not see any similar issue here in Github, are you aware of such an issue, anything we can do?

The login pop-up just stays like this after approving the connection, we never get authenticated anymore:
![image](https://user-images.githubusercontent.com/45175497/62221115-6b29de00-b3b1-11e9-8578-05eb35459f42.png)

Thanks,
Sergio

**Comment by shernandez-ctrlQS** [comment]
Hi Damian & team,

Attaching here the Console log in case it helps (references to our client´s data is removed).
[support-t.xxx.com-1564665510031.log](https://github.com/Strategery-Inc/jira-harvest-time-tracker/files/3456893/support-t.xxx.com-1564665510031.log)

Happy to give more info or jump into a phone call if needed.
Best,
Sergio

**Comment by VitaliiTsilnyk** [comment]
Hi Sergio,

In your log file there's an error that I've tracked down to a call to the `AJS.cookie.save()` method. Can you please open the console for that frozen pop-up window and execute `typeof AJS.cookie`?

**Comment by mfabris-ctrlqs** [comment]
Hello dear Vitalii 

The type of AJS.cookie is "undefined"

Here is the complete JS error

################################
```

A bad HTTP response code (404) was received when fetching the script.
batch.js?locale=en-US:44 GET https://support-t.xxxxx.com/plugins/servlet/com.strategery.jira-harvest/undefined/download/resources/com.infosysta.jira.in-app-and-desktop-notifications-for-jira:jira-notifier-resources-155/notification-settings.hb?_=1564744528773 404
send @ batch.js?locale=en-US:44
ajax @ batch.js?locale=en-US:44
E.<computed> @ batch.js?locale=en-US:44
nfj_getTemplateByName @ batch.js?locale=en-US:14715
nfj_fillProjectsElement @ batch.js?locale=en-US:14681
success @ batch.js?locale=en-US:14563
c @ batch.js?locale=en-US:44
fireWith @ batch.js?locale=en-US:44
u @ batch.js?locale=en-US:44
(anonymous) @ batch.js?locale=en-US:44
load (async)
send @ batch.js?locale=en-US:44
ajax @ batch.js?locale=en-US:44
nfj_fillData @ batch.js?locale=en-US:14548
njf_init @ batch.js?locale=en-US:14365
(anonymous) @ batch.js?locale=en-US:14361
(anonymous) @ batch.js?locale=en-US:276
setTimeout (async)
(anonymous) @ batch.js?locale=en-US:276
(anonymous) @ batch.js?locale=en-US:44
c @ batch.js?locale=en-US:44
fireWith @ batch.js?locale=en-US:44
ready @ batch.js?locale=en-US:44
H @ batch.js?locale=en-US:44
com.strategery.jira-harvest:base-resources.js:52 Uncaught (in promise) TypeError: Cannot read property 'save' of undefined
    at _0x4f1fc7.value (/s/14f26a8d1be550bb6464c693288b7841-CDN/agqf9y/802003/317a75b1fdeb21b48ce5fece3067a026/491e7fba184d786858d703d1339aa297/_/download/contextbatch/js/jira.browse.project,jira.view.issue,atl.general,jira.general,project.issue.navigator,jira.global,-_super/batch.js?agile_global_admin_condition=true&baseurl-check-resources=true&healthcheck-resources=true&jag=true&jaguser=true&jira.create.linked.issue=true&locale=en-US&richediton=true&whisper-enabled=true:26367)
    at window.<computed> (/s/14f26a8d1be550bb6464c693288b7841-CDN/agqf9y/802003/317a75b1fdeb21b48ce5fece3067a026/491e7fba184d786858d703d1339aa297/_/download/contextbatch/js/jira.browse.project,jira.view.issue,atl.general,jira.general,project.issue.navigator,jira.global,-_super/batch.js?agile_global_admin_condition=true&baseurl-check-resources=true&healthcheck-resources=true&jag=true&jaguser=true&jira.create.linked.issue=true&locale=en-US&richediton=true&whisper-enabled=true:26367)
    at _0xc324cb.<anonymous> (com.strategery.jira-harvest:base-resources.js:52)
    at r (com.strategery.jira-harvest:base-resources.js:28)
    at Generator._invoke (com.strategery.jira-harvest:base-resources.js:28)
    at Generator.e.<computed> [as next] (com.strategery.jira-harvest:base-resources.js:28)
    at _0x2b72b0 (com.strategery.jira-harvest:base-resources.js:52)
    at com.strategery.jira-harvest:base-resources.js:52
    at new Promise (<anonymous>)
    at _0xc324cb.<anonymous> (com.strategery.jira-harvest:base-resources.js:52)
value @ /s/14f26a8d1be550bb6464c693288b7841-CDN/agqf9y/802003/317a75b1fdeb21b48ce5fece3067a026/491e7fba184d786858d703d1339aa297/_/download/contextbatch/js/jira.browse.project,jira.view.issue,atl.general,jira.general,project.issue.navigator,jira.global,-_super/batch.js?agile_global_admin_condition=true&baseurl-check-resources=true&healthcheck-resources=true&jag=true&jaguser=true&jira.create.linked.issue=true&locale=en-US&richediton=true&whisper-enabled=true:26367
window.<computed> @ /s/14f26a8d1be550bb6464c693288b7841-CDN/agqf9y/802003/317a75b1fdeb21b48ce5fece3067a026/491e7fba184d786858d703d1339aa297/_/download/contextbatch/js/jira.browse.project,jira.view.issue,atl.general,jira.general,project.issue.navigator,jira.global,-_super/batch.js?agile_global_admin_condition=true&baseurl-check-resources=true&healthcheck-resources=true&jag=true&jaguser=true&jira.create.linked.issue=true&locale=en-US&richediton=true&whisper-enabled=true:26367
(anonymous) @ com.strategery.jira-harvest:base-resources.js:52
r @ com.strategery.jira-harvest:base-resources.js:28
(anonymous) @ com.strategery.jira-harvest:base-resources.js:28
e.<computed> @ com.strategery.jira-harvest:base-resources.js:28
_0x2b72b0 @ com.strategery.jira-harvest:base-resources.js:52
(anonymous) @ com.strategery.jira-harvest:base-resources.js:52
(anonymous) @ com.strategery.jira-harvest:base-resources.js:52
_0x2b598c @ com.strategery.jira-harvest:base-resources.js:52
_0x1fd2e7 @ com.strategery.jira-harvest:base-resources.js:52
a @ com.strategery.jira-harvest:base-resources.js:13
c @ com.strategery.jira-harvest:base-resources.js:13
d @ com.strategery.jira-harvest:base-resources.js:13
setTimeout (async)
o @ com.strategery.jira-harvest:base-resources.js:13
m @ com.strategery.jira-harvest:base-resources.js:13
requestAnimationFrame (async)
(anonymous) @ com.strategery.jira-harvest:base-resources.js:52
c @ batch.js?locale=en-US:44
fireWith @ batch.js?locale=en-US:44
ready @ batch.js?locale=en-US:44
H @ batch.js?locale=en-US:44
batch.js?locale=en-US:44 GET https://support-t.xxxx.com/plugins/servlet/com.strategery.jira-harvest/undefined/download/resources/com.infosysta.jira.in-app-and-desktop-notifications-for-jira:jira-notifier-resources-155/jira-notifier-template.hb?_=1564744528775 404

```

#############################

**Comment by VitaliiTsilnyk** [comment]
Looks like something interferes with the JS that our plugin requires. Can you try to disable the "In-App & Desktop Notifications for Jira" add-on? Looks like it's trying to load some resources on the page generated by our servlet and fails to do so.

**Comment by mfabris-ctrlqs** [comment]
Hi Vitalii

We have disabled the "In-App Notifications" plugin and tried with a clean browser. 

I log into JIRA
I go click the app button
I am asked to log in into harvest, which I do suffessfully
Then it starts again loading.

This time there is no 404 bad request, but the "TypeError: cannot read property 'save' of undefined" persists.


```
Downloading resources:

com.strategery.jira-harvest:base-resources.js:52 Uncaught (in promise) TypeError: Cannot read property 'save' of undefined
    at _0x4f1fc7.value (/s/d56396d5a4d5b03feacf890b5d9524ef-CDN/agqf9y/802003/317a75b1fdeb21b48ce5fece3067a026/65cd3a379a198bdb8cd8c2abbacdf6ed/_/download/contextbatch/js/jira.browse.project,jira.view.issue,atl.general,jira.general,project.issue.navigator,jira.global,-_super/batch.js?agile_global_admin_condition=true&baseurl-check-resources=true&healthcheck-resources=true&jag=true&jaguser=true&jira.create.linked.issue=true&locale=en-US&richediton=true&whisper-enabled=true:8704)
    at window.<computed> (/s/d56396d5a4d5b03feacf890b5d9524ef-CDN/agqf9y/802003/317a75b1fdeb21b48ce5fece3067a026/65cd3a379a198bdb8cd8c2abbacdf6ed/_/download/contextbatch/js/jira.browse.project,jira.view.issue,atl.general,jira.general,project.issue.navigator,jira.global,-_super/batch.js?agile_global_admin_condition=true&baseurl-check-resources=true&healthcheck-resources=true&jag=true&jaguser=true&jira.create.linked.issue=true&locale=en-US&richediton=true&whisper-enabled=true:8704)
    at _0xc324cb.<anonymous> (com.strategery.jira-harvest:base-resources.js:52)
    at r (com.strategery.jira-harvest:base-resources.js:28)
    at Generator._invoke (com.strategery.jira-harvest:base-resources.js:28)
    at Generator.e.<computed> [as next] (com.strategery.jira-harvest:base-resources.js:28)
    at _0x2b72b0 (com.strategery.jira-harvest:base-resources.js:52)
    at com.strategery.jira-harvest:base-resources.js:52
    at new Promise (<anonymous>)
    at _0xc324cb.<anonymous> (com.strategery.jira-harvest:base-resources.js:52)

```

###################

What can we do next?

@shernandez-ctrlQS

**Comment by mfabris-ctrlqs** [comment]
Forgot to mention: this is the one and only JS error reported in the console.

**Comment by VitaliiTsilnyk** [comment]
Is the `AJS.cookie` still undefined after the page finishes loading?

**Comment by mfabris-ctrlqs** [comment]
Yes! 

```
typeof AJS.cookie
"undefined"
```

**Comment by VitaliiTsilnyk** [comment]
I just released another version, 1.0.15-SNAPSHOT, can you please try it?

**Comment by mfabris-ctrlqs** [comment]
Thanks Vitalii, you killed it! 

The app is now working perfectly.
I did several tests and all looks fine now, actually it is even faster than before.
I also reenabled the app that gave the initial JS error and the Harvester app is working fine even with this other app enabled. 

We have the upgrade to live today at 15:00 (in 40 minutes) so the timing couldn't have been better. 
Awesome job, thanks a lot for the impeccable support.

Kind regards, greetings from Berlin!

Mirco, Sergio, ctlr QS team.
